wsgi.py

```diff
@@ -6,255 +6,348 @@
 import re
 import os
 from flask import Flask, redirect, url_for, request, Response
-#from flask_restful import reqparse
+from flask_httpauth import HTTPBasicAuth
+
+# from flask_restful import reqparse
 
 import sys
 
-from openshift_rolebindings import *
-from openshift_project import *
-from openshift_identity import *
-from openshift_user import *
+from openshift import *
 
 application = Flask(__name__)
+auth = HTTPBasicAuth()
 
-if __name__ != '__main__':
-    gunicorn_logger = logging.getLogger('gunicorn.error')
+if __name__ != "__main__":
+    gunicorn_logger = logging.getLogger("gunicorn.error")
     application.logger.handlers = gunicorn_logger.handlers
     application.logger.setLevel(gunicorn_logger.level)
 
 
-def get_user_token():
-    with open('/var/run/secrets/kubernetes.io/serviceaccount/token', 'r') as file:
+def get_openshift():
```

Comment:
You're currently calling this per-request, but the information retrieved here is static for the life of the container. It would be more effective to just assign the results of this to a global variable.

Reply:
Yes, you are right.  I was going to do it inside a singleton, but decided against that since this was easier to get working.  I should have left a TODO comment to signal my intent.

Reply:
Well, I am planning to replace this with a singleton.

I did it this way as it is brutally simple to understand and to get working.